main.tf

```
@@ -38,5 +39,8 @@ locals {
         "eks.amazonaws.com/role-arn" = module.addon-irsa[local.addon.name].iam_role_attributes.arn
       } : tomap({})
     }
+    extraArgs = concat(
+      one(var.irsa_assume_role_arns[*]) != null ? ["--aws-assume-role=${var.irsa_assume_role_arns[0]}"] : [],
```

```suggestion
      one(var.irsa_assume_role_arns[*]) != null ? ["--aws-assume-role=${one(var.irsa_assume_role_arns[*])}"] : [],
```
I would let it fail when more than 1 ARNs are passed to avoid confusion